### PR TITLE
HOSTEDCP-1600: [ARO-HCP] Add Cilium jobs

### PR DESCRIPTION
**Pull request by fxierh**
AKS MC + HC with Cilium as CNI. 

https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/55541/rehearse-55541-periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-cilium-guest-f7/1823588671098458112 (test phase killed manually)

### CODE REVIEW

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-cilium-guest-f7

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-cilium-guest-f7

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-cilium-guest-f7

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-cilium-guest-f7

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-byo-vnet-fips-guest-f7

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.17-amd64-nightly-azure-aks-hypershift-etcd-disk-encryption-guest-f7

**Comment by fxierh** [comment]
/cc @liangxia @heliubj18 

Could you please take a look ? Thank you

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-rosa-hcp-capi-private-stage-f7

File: ci-operator/step-registry/hypershift/azure/create/hypershift-azure-create-chain.yaml [hunk 1]
```diff
@@ -151,6 +158,24 @@ chain:
       oc wait --timeout=30m --for=condition=Available --namespace=clusters hostedcluster/${CLUSTER_NAME}
       echo "Cluster became available, creating kubeconfig"
       bin/hypershift create kubeconfig --namespace=clusters --name=${CLUSTER_NAME} >${SHARED_DIR}/nested_kubeconfig
+      
+      # In case of BYO CNI, the HC will never be ready before CNI components are up and running
+      if [[ $HYPERSHIFT_NETWORK_TYPE == "Other" || $HYPERSHIFT_NETWORK_TYPE == "Calico" ]]; then
```
Comment (heliubj18) [inline ci-operator/step-registry/hypershift/azure/create/hypershift-azure-create-chain.yaml:163]: is it same for calico cni?

Comment (fxierh) [reply ci-operator/step-registry/hypershift/azure/create/hypershift-azure-create-chain.yaml:163]: Yes. Calico has to be deployed manually as well.

Comment (heliubj18) [reply ci-operator/step-registry/hypershift/azure/create/hypershift-azure-create-chain.yaml:163]: /lgtm

**Pull request merged**